src/Chapter2.hs

```diff
@@ -606,8 +626,10 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
-
+--duplicate = error "duplicate: Not implemented!"
+duplicate x
+    | null x=[]
+    | otherwise=(head x):(head x):duplicate (tail x)
```

Comment:
Here it is also better to use pattern matching instead of using unsafe `head` function. (It is also more efficient 🙂 )